beets/dbcore/query.py

```diff
@@ -593,10 +607,27 @@ def find_date_and_format(string):
 
         if not string:
             return None
+
+        # Check for a relative date.
+        match_dq = re.match(cls.relative_re, string)
+        if match_dq:
+            sign = match_dq.group('sign')
+            quantity = match_dq.group('quantity')
+            timespan = match_dq.group('timespan')
+
+            # Add or subtract the given amount of time from the current
+            # date.
+            multiplier = -1 if sign == '-' else 1
+            days = cls.relative[timespan]
+            date = datetime.now() + \
+                timedelta(days=int(quantity) * days) * multiplier
+            string = date.strftime(cls.date_formats[5][0])
```

Comment:
@euri10: Turning things back into a string seems a little odd. Can we just `return cls(date, something)` here?